feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.spec.ts

```diff
@@ -68,4 +90,83 @@ describe('ConfigAttributeDropDownComponent', () => {
       })
     );
   });
+
+  it('should call emit of selectionChange onHandleQuantity', () => {
+    component.ownerKey = ownerKey;
+
+    spyOn(component.selectionChange, 'emit').and.callThrough();
+
+    component.onHandleQuantity(2);
+
+    expect(component.selectionChange.emit).toHaveBeenCalledWith(
+      jasmine.objectContaining({
+        changedAttribute: jasmine.objectContaining({
+          name: name,
+          uiType: Configurator.UiType.DROPDOWN,
+          groupId: groupId,
+          selectedSingleValue: component.attributeDropDownForm.value,
+        }),
+        ownerKey: ownerKey,
+        updateType: Configurator.UpdateType.ATTRIBUTE_QUANTITY,
+      })
+    );
+  });
+
+  it('should call onHandleQuantity of event onChangeQuantity', () => {
+    spyOn(component, 'onHandleQuantity');
+
+    const quantity = { quantity: 2 };
+
+    component.onChangeQuantity(quantity);
+
+    expect(component.onHandleQuantity).toHaveBeenCalled();
+  });
+
+  it('should call onSelect of event onChangeQuantity', () => {
+    spyOn(component, 'onSelect');
+
+    const quantity = { quantity: 0 };
+
+    component.onChangeQuantity(quantity);
+
+    expect(component.onSelect).toHaveBeenCalled();
+  });
+
+  it('should not display attribute quantity when dataType is no quantity', () => {
+    component.attribute.dataType = Configurator.DataType.USER_SELECTION_NO_QTY;
+
+    fixture.detectChanges();
+
+    CommonConfiguratorTestUtilsService.expectElementNotPresent(
+      expect,
+      htmlElem,
+      'cx-configurator-attribute-quantity'
+    );
+  });
+
+  it('should display attribute quantity when dataType is with attribute quantity', () => {
+    CommonConfiguratorTestUtilsService.expectElementPresent(
+      expect,
+      htmlElem,
+      '.form-group'
+    );
+  });
+
+  it('should allow quantity', () => {
+    expect(component.withQuantity).toBe(true);
+  });
+
+  it('should not allow quantity when service is missing ', () => {
+    component['quantityService'] = undefined;
```

Comment:
check my previous comment about this optional `quantityService`.

Reply:
picked

Reply:
done